reconstructSPI/iterative_refinement/expectation_maximization.py

```diff
@@ -476,14 +525,36 @@ def generate_slices(map_3d_f, xy_plane, rots):
         """
         n_rotations = len(rots)
         n_pix = len(map_3d_f)
-        slices = np.empty((n_rotations, n_pix, n_pix))
+        slices = np.empty((n_rotations, n_pix, n_pix), dtype=np.complex_)
```

Comment:
the dtype should be real (enforced) as the real and imaginary values are handled separately by calling `generate_slices` on them.

Reply:
Good catch. Later versions of scipy have map_coordinates able to handle complex values in the way that we want but that isn't true of older versions to my knowledge. There are still remnants of me trying to make use of this it seems!